Add docs link checker workflow

JKarlavige opened the pull request:
## What are you changing in this pull request and why?
[Asana](https://app.asana.com/0/1200099998847559/1203760149887730/f)

This adds a markdown link checker Git Action workflow to check markdown links on a schedule.

[Latest workflow run here](https://github.com/dbt-labs/docs.getdbt.com/actions/runs/5532859683)

A separate PR #3675 updates the `src` prop name to `path` for the `Snippet` and FAQ` prop. The link checker was picking up the `src` prop name and flagging those as broken links.

## TODO Before Merge
- [x] Update workflow to run on schedule.

JKarlavige committed:
add checker config

JKarlavige committed:
add excluded urls

JKarlavige committed:
add 2nd twitter exclude

JKarlavige commented:
Testing a few more things with the link checker, feel free to hold off on a review for now!

JKarlavige committed:
update faq prop name

JKarlavige committed:
update snippet prop name

JKarlavige committed:
trigger build

runleonarun reviewed (approved):
This looks great and we're so super excited you found the issues with broken links! Thanks so much, @JKarlavige!

JKarlavige commented:
> This looks great and we're so super excited you found the issues with broken links! Thanks so much, @JKarlavige!

Awesome! I'm thinking we can run this on a schedule to start analyzing how this workflow runs in production, i'm thinking once a day so there's a report to look at each morning.

From there, we can begin looking into sending alerts to Slack, or moving it into PR workflows and preventing merges unless the link checker passes. We can work this into a v2 Asana task once we're confident we have the link checker reporting accurate results.

If that's good with you, i will move forward with merging this in and getting it running in production!

JKarlavige committed:
Merge branch 'current' into feature/markdown-link-checker

JKarlavige committed:
update workflow name

JKarlavige committed:
update to run at 12:00 UTC every day

mirnawong1 committed:
Merge branch 'feature/markdown-link-checker' into update-component-prop-names

JKarlavige committed:
Merge branch 'current' into feature/markdown-link-checker

JKarlavige committed:
Update component prop names (#3675)

## What are you changing in this pull request and why?
This updates our existing `Snippet` and `FAQ` component `src` prop name
to `path`. The link checker in PR #3673 was flagging these components as
broken links due to the `src` prop. After updating the prop name to
`path`, this link check did not report those components as broken links.

[Workflow run from this
update](https://github.com/dbt-labs/docs.getdbt.com/actions/runs/5467755332)

## Previews
The following pages use the `Snippet` or `FAQ` components:
- Uses Snippet on 1st paragraph:
https://deploy-preview-3675--docs-getdbt-com.netlify.app/docs/introduction
- Has FAQ section:
https://deploy-preview-3675--docs-getdbt-com.netlify.app/docs/build/jinja-macros#faqs
- Discourse header is Snippet:
https://deploy-preview-3675--docs-getdbt-com.netlify.app/docs/build/incremental-models#questions-from-the-community
- FAQs:
https://deploy-preview-3675--docs-getdbt-com.netlify.app/docs/build/seeds#faqs

JKarlavige committed:
Merge branch 'current' into feature/markdown-link-checker
